frontend/src/lib/components/chart/chart-page/encoding/chart-encoding/encoding-components/MetricsEncodingMultiChoice.svelte

```diff
@@ -1,55 +1,53 @@
 <script lang="ts">
 	import { metrics } from '$lib/stores';
-	import { svelecteRenderer } from '$lib/util/util';
 	import Checkbox from '@smui/checkbox';
-	import Svelecte from 'svelecte';
 	import { createEventDispatcher } from 'svelte';
-	import { dndzone } from 'svelte-dnd-action';
-	import EncodingContainer from './EncodingContainer.svelte';
+	import { MultiSelect } from 'svelte-multiselect';
 
 	export let numberValues: number[];
 
 	const dispatch = createEventDispatcher<{ selected: number[] }>();
 
 	let options: { value: number; label: string }[] = [];
-	let value: number[] = [];
+	let value: { value: number; label: string }[] = [];
 
 	// initial options & values
 	options.push({ value: -1, label: 'slice size' });
 	$metrics.forEach((m) => {
 		options.push({ value: m.id, label: m.name });
 	});
-	value = numberValues;
+	value = numberValues.map((v) => {
+		return { value: v, label: options.find((o) => o.value === v)?.label || '' };
+	});
 
-	function updateDragOrder(val: number[]) {
-		// check if all elements are numbers (dndzone's place holder)
-		if (!val.some((i) => !Number.isInteger(i))) {
-			dispatch('selected', val);
-		}
+	function updateDragOrder(val: { value: number; label: string }[]) {
+		dispatch(
+			'selected',
+			val.map((v) => v.value)
+		);
 	}
 
 	$: updateDragOrder(value);
 </script>
 
-<EncodingContainer>
-	<div class="flex flex-col">
-		{#if value[0] != -2}
-			<Svelecte
-				style="width: 280px; flex:none;"
-				bind:value
-				{options}
-				{dndzone}
-				multiple={true}
-				placeholder="Select Metrics..."
-				renderer={svelecteRenderer}
-			/>
-		{/if}
-		<div class="ml-auto flex items-center">
-			<span>All Metrics</span>
-			<Checkbox
-				checked={value[0] == -2}
-				on:click={() => (value[0] === -2 ? (value = []) : (value = [-2]))}
-			/>
-		</div>
+<div class="flex flex-col">
+	{#if value.length === 0 || value[0].value != -2}
+		<MultiSelect
+			bind:selected={value}
+			{options}
+			liSelectedClass="!bg-primary-light ![&>svg]:fill-primary"
+			outerDivClass="!w-full !border-grey-light !py-1 !bg-white"
+			liActiveOptionClass="!bg-primary-light"
+		/>
+	{/if}
+	<div class="ml-auto flex items-center">
+		<span>All Metrics</span>
+		<Checkbox
+			checked={value.length > 0 && value[0].value == -2}
```

Comment:
I didn't do that either but maybe we should add a comment somewhere do explain this -2?